platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -0,0 +1,445 @@
+            soapfaultMessage = IOUtils.toString(soapfaultMessageStream);
+            securityHeader = IOUtils.toString(securityHeaderStream);
+            usernameToken = IOUtils.toString(userTokenStream);
+        } catch (IOException e) {
+            LOGGER.info("Unable to load templates for PAOS");
+        }
+
+    }
+
+    @Override
+    public void handleMessage(Message message) throws Fault {
+        List authHeader = (List) ((Map) message.getExchange()
+                .getOutMessage()
+                .get(Message.PROTOCOL_HEADERS)).get("Authorization");
+        String authorization = null;
+        if (authHeader != null && authHeader.size() > 0) {
+            authorization = (String) authHeader.get(0);
+        }
+        InputStream content = message.getContent(InputStream.class);
+        String contentType = (String) message.get(Message.CONTENT_TYPE);
+        if (contentType == null || !contentType.contains(APPLICATION_VND_PAOS_XML)) {
+            return;
+        }
+        try {
+            SOAPPart soapMessage = SamlProtocol.parseSoapMessage(IOUtils.toString(content));
+            Iterator iterator = soapMessage.getEnvelope()
+                    .getHeader()
+                    .examineAllHeaderElements();
+            IDPEntry idpEntry = null;
+            String relayState = "";
+            String responseConsumerURL = "";
+            String messageId = "";
+            while (iterator.hasNext()) {
+                Element soapHeaderElement = (SOAPHeaderElement) iterator.next();
+                if (RELAY_STATE.equals(soapHeaderElement.getLocalName())) {
+                    relayState = DOM2Writer.nodeToString(soapHeaderElement);
+                } else if (REQUEST.equals(soapHeaderElement.getLocalName())
+                        && soapHeaderElement.getNamespaceURI()
+                        .equals(URN_OASIS_NAMES_TC_SAML_2_0_PROFILES_SSO_ECP)) {
+                    try {
+                        soapHeaderElement = SamlProtocol.convertDomImplementation(
+                                soapHeaderElement);
+                        Request ecpRequest = (Request) OpenSAMLUtil.fromDom(soapHeaderElement);
+                        IDPList idpList = ecpRequest.getIDPList();
+                        if (idpList == null) {
+                            throw new Fault(new AccessDeniedException(
+                                    "Unable to complete SAML ECP connection. Unable to determine IdP server."));
+                        }
+                        List<IDPEntry> idpEntrys = idpList.getIDPEntrys();
+                        if (idpEntrys == null || idpEntrys.size() == 0) {
+                            throw new Fault(new AccessDeniedException(
+                                    "Unable to complete SAML ECP connection. Unable to determine IdP server."));
+                        }
+                        //choose the right entry, probably need to do something better than select the first one
```

Comment:
Could we make this configurable based on the requester hostname/port?  Should we write a ticket now to track that this should be configurable?